### PR TITLE
CI: add wait for sync tool

### PR DESCRIPTION
**Pull request by jordimassaguerpla**
## What does this PR change?

This script is used in the CI to wait for package synchronization in the mirror.

## GUI diff

No difference.

- [X] **DONE**

## Documentation
- No documentation needed
- [X] **DONE**

## Test coverage
- No tests

- [X] **DONE**

## Links

Fixes # https://github.com/SUSE/spacewalk/issues/18522

- [ ] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [X] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: susemanager-utils/testing/automation/wait-for-syncs.sh [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+set +ex
+host=""
+tries=60
+wait=10
+usage() {
+    echo "Usage: ${0} -h host [-t tries] [-w wait]"
+    echo "host is mandatory"
+    echo "If not specified, the default values are"
+    echo "tries: ${tries}"
+    echo "wait: ${wai}"
```
Comment (juliogonzalez) [inline susemanager-utils/testing/automation/wait-for-syncs.sh:11]:
```suggestion
    echo "wait: number of seconds to wait between retries (Optional, default value: ${wait})"
```

File: susemanager-utils/testing/automation/wait-for-syncs.sh [hunk 2]
```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+set +ex
+host=""
+tries=60
+wait=10
+usage() {
```
Comment (juliogonzalez) [inline susemanager-utils/testing/automation/wait-for-syncs.sh:6]:
```suggestion

usage() {
```

File: susemanager-utils/testing/automation/wait-for-syncs.sh [hunk 3]
```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
```
Comment (juliogonzalez) [inline susemanager-utils/testing/automation/wait-for-syncs.sh:1]:
```suggestion
#!/bin/bash +ex
```

File: susemanager-utils/testing/automation/wait-for-syncs.sh [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+set +ex
+host=""
+tries=60
+wait=10
+usage() {
+    echo "Usage: ${0} -h host [-t tries] [-w wait]"
+    echo "host is mandatory"
+    echo "If not specified, the default values are"
+    echo "tries: ${tries}"
```
Comment (juliogonzalez) [inline susemanager-utils/testing/automation/wait-for-syncs.sh:10]:
```suggestion
    echo "tries: number of attempts before failing (Optional, default value is ${tries})"
```

File: susemanager-utils/testing/automation/wait-for-syncs.sh [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+  sleep ${wait}
+done
+if [ ${i} -eq ${tries} ];then
+  echo "Reached max number of tries"
+  exit -1
+fi
+
+i=0
+rm -f sync-started
+echo "Waiting for end"
```
Comment (juliogonzalez) [inline susemanager-utils/testing/automation/wait-for-syncs.sh:60]:
```suggestion
echo "Waiting for the sync to end"
```

File: susemanager-utils/testing/automation/wait-for-syncs.sh [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+  i=$((${i}+1))
+  sleep ${wait}
+done
+if [ ${i} -eq ${tries} ];then
+  echo "Reached max number of tries"
+  exit -1
+fi
+
+i=0
+rm -f sync-started
```
Comment (juliogonzalez) [inline susemanager-utils/testing/automation/wait-for-syncs.sh:59]: If you use curl, I think you don't need this.

File: susemanager-utils/testing/automation/wait-for-syncs.sh [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+echo "Waiting for end"
+while [ $i -lt ${tries} ];do
+  wget -S http://${host}/sync-obs/sync-finished 2>/dev/null
```
Comment (juliogonzalez) [inline susemanager-utils/testing/automation/wait-for-syncs.sh:62]: Same as before, I recommend curl.

File: susemanager-utils/testing/automation/wait-for-syncs.sh [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+  fi
+  i=$((${i}+1))
+  sleep ${wait}
+done
+if [ ${i} -eq ${tries} ];then
+  echo "Reached max number of tries"
+  exit -1
+fi
+
+rm -f sync-finished
```
Comment (juliogonzalez) [inline susemanager-utils/testing/automation/wait-for-syncs.sh:75]: Not needed if you use curl.

File: susemanager-utils/testing/automation/wait-for-syncs.sh [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+  if [ $? -eq 0 ];then
+    echo "Sync finished"
+    break
+  fi
+  i=$((${i}+1))
+  sleep ${wait}
+done
+if [ ${i} -eq ${tries} ];then
+  echo "Reached max number of tries"
+  exit -1
```
Comment (juliogonzalez) [inline susemanager-utils/testing/automation/wait-for-syncs.sh:72]: This happens at different place, so maybe you want a different error code.

File: susemanager-utils/testing/automation/wait-for-syncs.sh [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+while [ $i -lt ${tries} ];do
+  wget -S http://${host}/sync-obs/sync-finished 2>/dev/null
+  if [ $? -eq 0 ];then
+    echo "Sync finished"
+    break
+  fi
+  i=$((${i}+1))
+  sleep ${wait}
+done
+if [ ${i} -eq ${tries} ];then
```
Comment (juliogonzalez) [inline susemanager-utils/testing/automation/wait-for-syncs.sh:70]:
```suggestion
if [ ${i} -eq ${tries} ]; then
```

File: susemanager-utils/testing/automation/wait-for-syncs.sh [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+  echo "Reached max number of tries"
+  exit -1
+fi
+
+rm -f sync-finished
+
```
Comment (juliogonzalez) [inline susemanager-utils/testing/automation/wait-for-syncs.sh:76]:
```suggestion
```

**Review by juliogonzalez — COMMENTED** [review]
Mostly style, but I'd seriously considering using `curl` instead of `wget`

File: susemanager-utils/testing/automation/wait-for-syncs.sh [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+shift $((OPTIND-1))
+if [ "${host}" == "" ];then
+    echo "ERROR: Please specify host"
+    exit -3
+fi
+echo "Wait for sync to start and finish"
+i=0
+echo "Waiting for start"
+while [ $i -lt $tries ];do
+  wget -S http://${host}/sync-obs/sync-started 2>/dev/null
```
[the remark this answers is not in the document]
Comment (jordimassaguerpla) [reply susemanager-utils/testing/automation/wait-for-syncs.sh:45]: With `curl` I can't tell with `$?` if the file was available:

```
jordi@linux-9y1l:~$ curl -s http://minima-mirror.mgr.suse.de/sync-obs/sync-finished
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en">
<head>
<title>Object not found!</title>
<link rev="made" href="mailto:%5bno%20address%20given%5d" />
<style type="text/css"><!--/*--><![CDATA[/*><!--*/ 
    body { color: #000000; background-color: #FFFFFF; }
    a:link { color: #0000CC; }
    p, address {margin-left: 3em;}
    span {font-size: smaller;}
/*]]>*/--></style>
</head>

<body>
<h1>Object not found!</h1>
<p>


    The requested URL was not found on this server.

  

    If you entered the URL manually please check your
    spelling and try again.

  

</p>
<p>
If you think this is a server error, please contact
the <a href="mailto:%5bno%20address%20given%5d">webmaster</a>.

</p>

<h2>Error 404</h2>
<address>
  <a href="/">minima-mirror.mgr.suse.de</a><br />
  <span>Apache</span>
</address>
</body>
</html>

jordi@linux-9y1l:~$ echo $?
0

```

**Comment by jordimassaguerpla** [comment]
Thanks for your revision. I have fixed it based on your suggestions.

**Pull request merged**